helper/resource/testing_new.go

```diff
@@ -0,0 +1,188 @@
+package resource
+
+import (
+	"fmt"
+	"log"
+	"reflect"
+	"strings"
+	"testing"
+
+	tftest "github.com/apparentlymart/terraform-plugin-test"
+	"github.com/davecgh/go-spew/spew"
+	tfjson "github.com/hashicorp/terraform-json"
+	"github.com/hashicorp/terraform-plugin-sdk/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func getState(t *testing.T, wd *tftest.WorkingDir) *terraform.State {
+	jsonState := wd.RequireState(t)
+	state, err := shimStateFromJson(jsonState)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return state
+}
+
+func RunLegacyTest(t *testing.T, c TestCase, providers map[string]terraform.ResourceProvider) {
```

Comment:
Why is this function called `Legacy` when it's the one that calls binary Terraform? 🤔  Shouldn't it be called something like `RunNewTest`?

Reply:
This was an artifact from me if I recall. I think my mindset was there would be a future where we have a NEW testing framework/API, and this function is called to run tests defined in our "old" API (the current one, resource.TestCase etc).

Reply:
Probably too forward thinking haha.

Reply:
I'll change this to `RunNewTest` to match `testing_new.go`, which I think is helpful naming while the SDK contains both the legacy and new test frameworks.

When the legacy test framework is removed, we can change these names as part of the major version bump.